File: Corte1/Python/Ejercicio5/Ejercicio5.py
class Numeros:
    def __init__(self, lista):
        self.lista = lista

    def valor_maximo(self):
        return max(self.lista)

    def contar_pares(self):
        return sum(1 for n in self.lista if n % 2 == 0)


n1 = Numeros([1, 2, 3, 4, 5, 6])
n2 = Numeros([7, 9, 11, 13])
n3 = Numeros([10, 20, 30, 40])

print(n1.valor_maximo(), n1.contar_pares())
print(n2.valor_maximo(), n2.contar_pares())
print(n3.valor_maximo(), n3.contar_pares())

File: Corte1/Java/Ejercicio5/Ejercicio5.java
class Numeros {
    private int[] lista;

    public Numeros(int[] lista) {
        this.lista = lista;
    }

    public int valorMaximo() {
        int max = lista[0];
        for (int n : lista) {
            if (n > max) {
                max = n;
            }
        }
        return max;
    }

    public int contarPares() {
        int contador = 0;
        for (int n : lista) {
            if (n % 2 == 0) {
                contador++;
            }
        }
        return contador;
    }
}

public class Main {
    public static void main(String[] args) {
        Numeros n1 = new Numeros(new int[]{1, 2, 3, 4, 5, 6});
        Numeros n2 = new Numeros(new int[]{7, 9, 11, 13});
        Numeros n3 = new Numeros(new int[]{10, 20, 30, 40});

        System.out.println(n1.valorMaximo() + " " + n1.contarPares());
        System.out.println(n2.valorMaximo() + " " + n2.contarPares());
        System.out.println(n3.valorMaximo() + " " + n3.contarPares());
    }
}
